Extend upselling extension to show more than one slider

AienTech opened the pull request:
# Pull Request Template

## Description

add the following changes to the extension:

- add new position to upselling

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I updated the CHANGELOG.md

Commit:
CCP-2460: Extend upselling extension to show more than one slider

- add new position to upselling

Commit:
CCP-2460: array of configs

- add the tweaks to make it able to give more than one configs to show the related products

Commit:
CCP-2460: fix the missing readme

Commit:
CCP-2460: fix an issue with no properties in relations

Commit:
CCP-2460: fix the issue with products not getting fetched when tpe is set to property

Carsten89 commented on frontend/components/Slider/index.jsx:
```diff
@@ -51,12 +54,14 @@ class Slider extends Component {
-    }));
+  componentDidMount () {
+    const { dispatch } = this.props
+    if (this.props.type !== "property") {
```
// TODO: move to some common place
const TYPE_PROPERTY = 'property';

```suggestion
    if (this.props.type !== TYPE_PROPERTY) {
```

Carsten89 commented on frontend/components/Slider/index.jsx:
```diff
@@ -119,6 +124,14 @@ const mapStateToProps = (state, props) => {
     type: props.type,
   };
+
+  if (params.type === 'property') {
```
```suggestion
  if (params.type === 'TYPE_PROPERTY') {
```
do the same for all other places

Carsten89 commented:
Please fix the ESlint error

Commit:
CCP-2460: fix lint issues

Commit:
CCP-2460: remove magical string 'property'

Commit:
CCP-2460: add property type to constants.js

Carsten89 commented on frontend/components/PDPSlider/index.jsx:
```diff
@@ -12,8 +12,8 @@ const PDPSlider = ({ productId, config }) => {
 
-  if (config.length === 0) {
-    return null
+  if (!config) {
```
since your default prop is `config: {}`, this check will always be false because 
empty object evaluates to true

Carsten89 commented on frontend/portals/ProductDetailPageSlider/index.jsx:
```diff
@@ -13,16 +13,22 @@ const { productPage } = getConfig();
+    return <PDPSlider key={`key_${idx}_${config.type}`} config={config} />;
+  });
+
+  return <div>{sliders}</div>;
```
sliders can be empty array. and it will render `<div></div>`
only render the divs if there are sliders

Carsten89 commented on frontend/portals/ProductDetailPageSlider/index.jsx:
```diff
@@ -13,16 +13,22 @@ const { productPage } = getConfig();
-  if (name !== productPage.position) {
-    return null;
-  }
+  const configs = Array.isArray(productPage) ? productPage : [productPage];
```
this should not happen inside the component since it will be executed on every render, but does not use anything from the component. Please move to line 7

Carsten89 commented:
please fix the tests

Carsten89 commented on README.md:
```diff
@@ -11,17 +11,20 @@ Currently it's possible to configure a slider which is rendered on the Product D
         "showName": "Boolean. If not true, no product name is shown",
         "nameLines": "Number. Maximum lines item name should be possible. If empty defaults to 2",
-    }
+        "property": "(optional) can be used when type is set to 'property', refers to the product property to show the related products with. Product ids needs to be comma seperated ids of the products related (e.g \"1,, 2,3 \")"
```
```suggestion
        "property": "(optional) can be used when type is set to 'property', refers to the product property to show the related products with. Product ids needs to be comma seperated ids of the products related (e.g \"1,2,3\")"
```

Carsten89 commented on frontend/selectors/index.js:
```diff
@@ -56,3 +60,43 @@ export const hasProductRelationsFiltered = params => createSelector(
+
+    const relatedProperty = properties.find(property => property.label === propertyLabel);
+
+    if (relatedProperty) {
```
We prefer something like this in our code because of readability:
```
if (!relatedProperty) {
 return []
}
      const { value } = relatedProperty;
      return value
        .split(',')
        .filter(Boolean)
        .map(s => s.trim());
    }
```

Carsten89 commented on frontend/selectors/index.js:
```diff
@@ -56,3 +60,43 @@ export const hasProductRelationsFiltered = params => createSelector(
+  }
+);
+
+export const getProductsFromProperty = createSelector(
```
```suggestion
export const getProductsDataFromProperty = createSelector(
```

Carsten89 commented on frontend/selectors/index.js:
```diff
@@ -56,3 +60,43 @@ export const hasProductRelationsFiltered = params => createSelector(
   productIds => productIds && productIds.length
 );
+
+export const getProductRelationsFromProperty = createSelector(
```
```suggestion
export const getProductRelationsIdsFromProperty = createSelector(
```

Commit:
CCP-2460: fix linting issues

Commit:
CCP-2460: refactor function naming

Commit:
CCP-2460: tests

- update tests and snapshots
- use a filter function to filter falsy configs in ProductDetailPageSlider/index.jsx
- add a backward compatible test

Commit:
CCP-2460: tests

- update tests and snapshots
- use a filter function to filter falsy configs in ProductDetailPageSlider/index.jsx
- add a backward compatible test

Carsten89 commented:
Test are failing
![image](https://user-images.githubusercontent.com/5811286/128182390-6674559c-de7b-4856-bcbc-f8f2fdb66ba1.png)
